### DISCUSSION TITLE
Client adapter for Mysql

### DISCUSSION THREAD

**Discussion by m-shariq** [prisma/prisma#24608] [Q&A]
### Question

Hi,

I see that there are client adapters available for PostgreSQL, PlanetScale, and D1 for passing custom clients. My question is, do we have any existing adapter available for MySQL? I'm currently working on my own adapter and need some information regarding one specific issue.

Looking forward to your reply.

Thank you.

**Comment by SilasRodrigues19** [comment]
Hey @m-shariq 

I believe this is what you are looking for:

https://www.prisma.io/docs/orm/overview/databases/mysql

**Reply by m-shariq** [reply]
No, i am developing something like `@prisma/adapter-pg` for mysql. But the issue is resolved. Thanks for the reply.

**Comment by ludralph** [comment]
Hi @m-shariq 👋 

Thank you for raising this question.

What’s the usecase here? We do support Planetscale driver, which is MySQL. The purpose of driver-adapters is to allow us to run where native databases/nodejs can not. In that sense, doing, for example mysql2 adapter right now does not make much sense since there is native support for MySQL already, but we can always reconsider if we have strong usecase for this.

**Reply by m-shariq** [reply]
Hi, @ludralph,

I am working on using the same MySQL pool for Knex and Prisma as I am incrementally migrating from Knex to Prisma. I have connection limitations on my DB server, so the same connection pool needs to be used. This is possible for Knex as I can pass my custom connection pool, but it is not possible for Prisma with MySQL.

Thanks

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!

**Reply by medz** [reply]
> [prisma-orm-manifesto](https://www.prisma.io/blog/prisma-orm-manifesto).

It is because of prisma-orm-manifesto that we should reopen it. From the progress of the past six months, I can see that Prisma ORM 7 is more dependent on adapters, and wasm engine cannot run directly on edge or serverless (it must rely on adapters)

Not all users are planetscale users. You should know that planetscale is a paid service. Now clouaflare Hyperdrive provides a free cluster proxy pool (supporting PG/MySQL). You can directly use traditional MySQL for edge networks.

Just like there is a pg driver now, you can directly use it to connect to the traditional pg database location.

On the contrary, I think closing this discussion is a violation of prisma-orm-manifesto. Prisma has not disclosed the support plan for first-class databases to the public anywhere. But from the manifesto, it is certain to support first-class databases. So does MySQL not belong to the category of first-class databases?

This has nothing to do with the squeeze issue. This is something Prisma itself promised to the outside world. You should know that prisma-orm-manifesto has caused a lot of trouble for Prisma users. Especially the current development state of Prisma is very slow and has many regression issues.

Isn't it the official reason of Prisma that the MySQL adapter issue was squeezed? Other insignificant or unconfirmed issues can be closed. But this is a functional issue that Prisma promised.

**Comment by alisd23** [comment]
Hi @ludralph, I assume your comment from 2024 is not relevant anymore as the new [rust-free](https://www.prisma.io/docs/orm/prisma-client/setup-and-configuration/no-rust-engine#prisma-orm-without-rust-engines) setup **requires** an adapter? Clearly the *planetscale* adapter is not suited for regular MySQL databases as it has a lot of planetscale-specific functionality.

We are also using MySQL and I'm a little confused on why MySQL seems to be the only major database skipped? It is far more popular than Postgresql: https://db-engines.com/en/ranking

Can you please explain how we are supposed to use a regular MySQL with Prisma in this new rust-free setup?

Thanks

**Reply by alisd23** [reply]
Just found this roadmap: https://github.com/prisma/prisma/issues/26592

Still unclear on when MySQL will be supported and no PR up yet, so it probably won't be supported for the near future. Back to the rust version we go...

**Comment by alisd23** [comment]
Follow-up: this is, as of very recently, up in a PR 👍 

https://github.com/prisma/prisma/pull/27393
